modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,374 @@
+    outputName = static_cast<std::string>(objPtr.getPropertyValue("OutputName"));
+
+    assert(blockSizeMs > 0);
+    assert(classCount > 0);
+
+}
+
+FunctionBlockTypePtr ClassifierFbImpl::CreateType()
+{
+    return FunctionBlockType("ref_fb_module_classifier", "Classifier", "Signal classifing");
+}
+
+void ClassifierFbImpl::processSignalDescriptorChanged(const DataDescriptorPtr& inputDataDescriptor,
+                                                   const DataDescriptorPtr& inputDomainDataDescriptor)
+{
+    if (inputDataDescriptor.assigned())
+        this->inputDataDescriptor = inputDataDescriptor;
+    if (inputDomainDataDescriptor.assigned())
+        this->inputDomainDataDescriptor = inputDomainDataDescriptor;
+
+    configure();
+}
+
+void ClassifierFbImpl::configure()
+{
+    if (!inputDataDescriptor.assigned() || !inputDomainDataDescriptor.assigned())
+    {
+        LOG_D("Incomplete signal descriptors")
+        return;
+    }
+
+    try
+    {
+        if (inputDataDescriptor.isStructDescriptor() || inputDataDescriptor.getDimensions().getCount() > 0)
+            throw std::runtime_error("Incompatible input value data descriptor");
+
+        inputSampleType = inputDataDescriptor.getSampleType();
+        if (inputSampleType != SampleType::Float64 &&
+            inputSampleType != SampleType::Float32 &&
+            inputSampleType != SampleType::Int8 &&
+            inputSampleType != SampleType::Int16 &&
+            inputSampleType != SampleType::Int32 &&
+            inputSampleType != SampleType::Int64 &&
+            inputSampleType != SampleType::UInt8 &&
+            inputSampleType != SampleType::UInt16 &&
+            inputSampleType != SampleType::UInt32 &&
+            inputSampleType != SampleType::UInt64)
+            throw std::runtime_error("Invalid sample type");
+
+        if (inputDomainDataDescriptor.getSampleType() != SampleType::Int64 && inputDomainDataDescriptor.getSampleType() != SampleType::UInt64)
+        {
+            throw std::runtime_error("Incompatible domain data sample type");
+        }
+
+        const auto domainRule = inputDomainDataDescriptor.getRule();
+        if (domainRule.getType() != DataRuleType::Linear)
+        {
+            throw std::runtime_error("Domain rule type is not Linear");
+        }
+
+        const auto domainRuleParams = domainRule.getParameters();
+
+        inputDeltaTicks = domainRuleParams.get("delta");
+
+        auto outputDataDescriptorBuilder = DataDescriptorBuilder().setSampleType(SampleType::Float64);
+        
+        auto dimensions = List<IDimension>();
+        if (useExplicitDomain) 
+            dimensions.pushBack(Dimension(ListDimensionRule(explicitDimension)));
+        else 
+        {
+            auto inputLowValue = static_cast<Float>(inputDataDescriptor.getValueRange().getLowValue());
+            auto inputHighValue = static_cast<Float>(inputDataDescriptor.getValueRange().getHighValue());
```

Comment:
Not all descriptors might have this value configured. In that case, the signal should be dropped, or a default low/high should be present.